card/db/migrate_core_cards/20150429090551_search_card_context.rb

```diff
@@ -0,0 +1,25 @@
+# -*- encoding : utf-8 -*-
+
+class SearchCardContext < Card::CoreMigration
+  def up
+    sep = %r{\W}
+    replace = [
+      ['[lr]+','l\\1'],
+      ['[LR]+','L\\1'],
+      ["(?=[lrLR]+#{sep})(?=[LR]*[lr]+)(?=[lr]*[LR]+).*",'l\\1'],   # mix of lowercase and uppercase l's and r's
+      ['left',   'LL'],
+      ['right',  'LR'],
+      ['self',   'left'],
+      ['',       'left'],
+    ]
+    Card.search(:type=>'search').each do |card|
+      if card.cardname.junction?
+        content = card.content
+        replace.each do |key, val|
+          content.gsub!(/(?<=#{sep})_(#{key})(?=#{sep})/, "_#{val}")
```

Comment:
Good that you asked. While explaining it I noticed that `.*` in line 9 is too greedy. It should say `[lrLR]+` instead